acados/ocp_nlp/ocp_nlp_sqp_rti.c

```diff
@@ -546,51 +688,93 @@ int ocp_nlp_sqp_rti(void *config_, void *dims_, void *nlp_in_, void *nlp_out_,
 #endif
 
     // initialize QP
-    ocp_nlp_initialize_qp(config, dims, nlp_in, nlp_out, nlp_opts, nlp_mem, nlp_work);
-
-        
+    ocp_nlp_initialize_qp(config, dims, nlp_in, nlp_out,
+        nlp_opts, nlp_mem, nlp_work);
 
     /* SQP body */
 	int sqp_iter = 0;
 	nlp_mem->sqp_iter = &sqp_iter;
 
     // linearizate NLP and update QP matrices
     acados_tic(&timer1);
-    ocp_nlp_approximate_qp_matrices(config, dims, nlp_in, nlp_out, nlp_opts, nlp_mem, nlp_work);
-    mem->time_lin += acados_toc(&timer1);
+    ocp_nlp_approximate_qp_matrices(config, dims, nlp_in,
+        nlp_out, nlp_opts, nlp_mem, nlp_work);
 
-    // update QP rhs for SQP (step prim var, abs dual var)
-    ocp_nlp_approximate_qp_vectors_sqp(config, dims, nlp_in, nlp_out, nlp_opts, nlp_mem, nlp_work);
+    mem->time_lin += acados_toc(&timer1);
 
     // regularize Hessian
     acados_tic(&timer1);
-    config->regularize->regularize_hessian(config->regularize, dims->regularize, opts->nlp_opts->regularize, nlp_mem->regularize_mem);
+    config->regularize->regularize_hessian(config->regularize,
+        dims->regularize, opts->nlp_opts->regularize, nlp_mem->regularize_mem);
+
     mem->time_reg += acados_toc(&timer1);
+}
+
+int ocp_nlp_sqp_rti_feedback_step(void *config_, void *dims_,
+    void *nlp_in_, void *nlp_out_, void *opts_, void *mem_, void *work_)
+{
+    acados_timer timer1;
+
+    ocp_nlp_dims *dims = dims_;
+    ocp_nlp_config *config = config_;
+    ocp_nlp_sqp_rti_opts *opts = opts_;
+    ocp_nlp_opts *nlp_opts = opts->nlp_opts;
+    ocp_nlp_sqp_rti_memory *mem = mem_;
+    ocp_nlp_in *nlp_in = nlp_in_;
+    ocp_nlp_out *nlp_out = nlp_out_;
+    ocp_nlp_memory *nlp_mem = mem->nlp_mem;
+    ocp_qp_xcond_solver_config *qp_solver = config->qp_solver;
+
+    ocp_nlp_sqp_rti_workspace *work = work_;
+    ocp_nlp_sqp_rti_cast_workspace(config, dims, opts, mem, work);
+    ocp_nlp_workspace *nlp_work = work->nlp_work;
+
+    int qp_iter = 0;
+    int qp_status = 0;
+	double tmp_time;
+
+    // linearizate NLP and update QP matrices
+    acados_tic(&timer1);
```

Comment:
Actually, there is no linearization at all in the feedback phase. I have removed the comment. Moreover, since initial value embedding + vectors update should be very fast, I would not add a separate timing entry for it. It will only count towards the total_time computation. (b01ff2a)